### PR TITLE
calculate last bit in myers1999_block once

### PR DESCRIPTION
**Pull request by maxbachmann**
Similar to myers1999_simple this calculates the index of the last bit only once.
On my system this improves the runtime by approximately 10%.

### CODE REVIEW

**Comment by fujimotos** [comment]
@maxbachmann I took a look at your patch. It looks good to me.

FWIW, before merging your work, I am considering to migrate polyleven
to a more standard licensing model.

 * The current license model depends on the fact that I'm the
   single author of polyleven ("I have all the copyrights on this
   work, and I abandon my rights")

 * However, once I accept other people's contribution, it does not
   work like that anymore.

The following is what I am thinking to do:

 * Apply MIT License to polyleven from v0.7 (v0.6 is the last version released
   under public domain)

 * Add Max Bachmann (you) as the co-author to `LICENSE`.

 * In other words, polyleven becomes a joint work of two authors,
   each agreeing to release their work under MIT license.

This migration seems to be good for two reasons: 1) it leaves little uncertainty
regarding the license status, and 2) it also gives you an explicit acknowledgement
for your contribution

If you are fine with the above, please insert your name to `LICENSE` as follows:

```
Copyright (c) 2021 Fujimoto Seiji <fujimoto@ceptord.net>
Copyright (c) 2021 Max Bachmann <kontakt@maxbachmann.de>

Permission is hereby granted, free of charge, ...
```

After you update this PR accordingly, I'm going to merge your patch.

**Comment by maxbachmann** [comment]
@fujimotos sure the MIT License is fine with me

**Comment by fujimotos** [comment]
@maxbachmann Thank you. I just merged your PR.
